### ISSUE TITLE
feat(autocomplete): add support for aria-labelledby

### ISSUE THREAD

**Issue by katiehockman** [angular/material#10815]
**Actual Behavior**:
 -  `What is the issue? *`
The aria-label and placeholder text are the same in situations where no floatingLabel is defined with md-autocomplete. This means that some screen readers will read off the placeholder text twice (once for the placeholder, and once for the aria-label).
For example:
`<md-autocomplete placeholder="What is your favorite city?"/>`
will create
`<input aria-label="What is your favorite city?" placeholder="What is your favorite city?">`

 -  `What is the expected behavior?`
There should always be a visible label for a combobox, thus there should not be a situation where the placeholder text acts as the label. However, given the constraint that this implementation is already in production, this is the proposed change:
Add an md-aria-labelledby attribute to md-autcomplete. If defined, then return an <input> element that has an aria-labelledby attribute in the place of the aria-label attribute that contains the id of the visible label.
For example:
`<label id="cityLabel">What is your favorite city?</label>`
`<md-autocomplete placeholder="Enter city name" md-aria-labelledby="cityLabel"/>`
would create
`<input ... aria-labelledby="cityLabel" placeholder="Enter city name" >`
Now, screen readers will always read "What is your favorite city?" as the label for the combobox, and some will read the placeholder text (which is extraneous information).
**Solution that will not work:** Removing the aria-label from the <input> element given the current implementation. There always needs to be a label for this combobox, and one cannot rely on a placeholder as a label since many assistive technologies will not read the placeholder. The solution outlined above is a workaround because replacing the aria-label with an aria-labelledby in all cases will be a breaking change for some users.

**CodePen** (or steps to reproduce the issue): *
https://material.angularjs.org/1.1.4/demo/autocomplete#basic-usage
Using Voiceover, navigate to the autocomplete combobox in the Basic Usage section. Notice that "What is your favorite UX state?" is read twice.

**AngularJS Versions**: *
 -  `AngularJS Version:`
 -  `AngularJS Material Version:` 1.1.4

**Additional Information**:
 -  `Browser Type: *` Chrome
 -  `Browser Version: *` Version 58.0.3029.110 (64-bit)
 -  `OS: *` macOS Sieera Version 10.12.5
 -  `Stack Traces:`
 -  `Screen Reader:` Voiceover (not reproducible with ChromevoxNext)

----

**Comment by Splaktar** [comment]
This is specifically related to using `md-autocomplete` when `md-floating-label` is not defined. 

This configuration appears to no longer be aligned with the latest version of the [Material Design spec](https://material.io/design/components/text-fields.html#anatomy) as all inputs within the updated spec have floating labels. Of course support for this won't be removed from AngularJS Material.

Using Chrome Version 68.0.3440.84 (Official Build) (64-bit) on macOS High Sierra 10.13.6 (17G65) with VoiceOver, I am not able to reproduce this repeated reading of the `aria-label` and `placeholder` with AngularJS Material 1.1.4 or 1.1.10. VoiceOver appears to have been updated to avoid reading both.

I am also not able to reproduce it with Chromevox on ChromeOS 68. 

I am however able to add support for `aria-labelledby` to `md-autocomplete` if that is still a useful enhancement. Please let me know.

**Comment by katiehockman** [comment]
I'm also not able to reproduce this anymore, it's very possible that Voiceover or Chrome has fixed this issue, where it detects identical placeholder and label text to only read one of them.
I do still think that having aria-labelledby as an option for users would be helpful, though I don't have all of the context with what has changed with the MD spec or Angular lately. Ideally there isn't too much redundancy of text within the element. So, what I wouldn't recommend is creating something like this:

```
<label id="cityLabel">What is your favorite city?</label>
<input ... aria-labelledby="cityLabel" aria-label="Enter city name" placeholder="Enter city name" >
```

That will still have the same problem of having "Enter city name" twice in the element, and could cause issues with assistive technology, despite the fact that this problem appears to have been fixed with Voiceover + Chrome. I would only recommend removing aria-label in the case where another label is clearly provided (e.g. by defining an aria-labelledby attribute).
